src/elements/forms/text-field.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import textFieldClassNames from './text-field.css'
+
+import { classNames } from 'utils'
+
+const generateId = () =>
+  // Math.random should be unique because of its seeding algorithm.
+  // Convert it to base 36 (numbers + letters), and grab the first 9 characters
+  // after the decimal.
+  Math.random()
+    .toString(36)
+    .substr(2, 9)
+
+const TextField = ({
+  children,
+  id,
+  className,
+  label,
+  size,
+  tag: Tag = 'div',
+  ...inputProps
+}) => {
+  const controlId = [id, 'control'].join('-')
+
+  return (
+    <Tag
+      id={id}
+      className={classNames
+        // prevents passing 'default' to the class list
+        .use('container', { [size]: size !== 'default' }, className)
```

Comment:
There can be a collision. When className prop contains for example `container` or whatever class defined in `text-fields.css`. I think you are aware of it but would be nice to keep it in mind and solve it somehow. Not necessary here and now. 

```jsx
<TextField
  id="login-password"
  type="password"
  name="password"
  label="Password"
  className="container"
  placeholder="🔒 Something secure"
/>
```

Current output:

```
<div class="text-field-container-1W0Hy text-field-container-1W0Hy">...</div>
```

Expected output:
```
<div class="text-field-container-1W0Hy container">...</div>
```

Reply:
I am aware bu I cannot find convenient solution. The best what I can find is either

```js
classNames.use(...classes).from(module).add(customClasses)
```

or process chain differently, i.e. do not do perform it in `toString()` but do it right in the method.

Any suggestions?

Reply:
I don't know. I think you can leave it as is till we run into this problem. You probably won't like it since it's YAGNI principle I guess. If you want to fix it then I vote for the first solution.

Reply:
Not really in this case. Could you please create an issue for this?